js/tji-datasets-explore.js

```diff
@@ -1,15 +1,349 @@
+    scales: {
+      yAxes: [{
+        ticks: {
+          beginAtZero:true
+        }
+      }]
+    }
+  }
+  return _.extend(options, this.get_options_overrides());
+}
+
+TJIGroupByBarChart.prototype.get_options_overrides = function() {
+  return {};
+}
+
+TJIGroupByBarChart.prototype.get_group_counts = function(data) {
+  data = _.filter(data, this.groupBy);
+  var grouped = _.groupBy(data, this.groupBy);
+  var keys = _.sortBy(_.keys(grouped));
+  if (keys.indexOf(this.missing_data_label) !== -1) {
+    keys.splice(keys.indexOf(this.missing_data_label), 1);
+    keys.push(this.missing_data_label);
+  }
+  var counts = _.map(keys, function(k){ return grouped[k].length});
+  return {
+    keys: keys,
+    counts: counts
+  };
+}
+
+
+// ********************************************************************
+// * Extends TJIGroupByBarChart... but for doughnuts.
+// ********************************************************************
+
+
+var TJIGroupByDoughnutChart = function(elt_id, groupBy, data, missing_data_label) {
+  TJIGroupByBarChart.call(this, elt_id, groupBy, data, missing_data_label);
+}
+TJIGroupByDoughnutChart.prototype = Object.create(TJIGroupByBarChart.prototype);
+TJIGroupByDoughnutChart.prototype.constructor = TJIGroupByDoughnutChart;
+TJIGroupByDoughnutChart.prototype.type = 'doughnut';
+TJIGroupByDoughnutChart.prototype.color_palette = [
+  COLOR_TJI_BLUE, COLOR_TJI_RED, COLOR_TJI_DEEPBLUE, COLOR_TJI_PURPLE,
+  COLOR_TJI_YELLOW, COLOR_TJI_TEAL, COLOR_TJI_DEEPRED, COLOR_TJI_DEEPPURPLE,
+];
+TJIGroupByDoughnutChart.prototype.get_options_overrides = function() {
+  return {
+    scales: {},
+    legend: {
+      display: true,
+      position: 'bottom',
+      labels: {
+        fontSize: 12,
+      }
+    },
+    pieceLabel: {
+      mode: function (args) {
+        return args.percentage + '%';
+      },
+      precision: 0,
+      showZero: true,
+      fontSize: 10,
+      fontColor: '#fff',
+      // available value is 'default', 'border' and 'outside'
+      position: 'default'
+    }
+  };
+}
+
+
+// ********************************************************************
+// * "Class" that creates and manages all charts and the filter panel.
+// *
+// * Constructor arguments:
+// *   chart_configs: array of chart config objects,
+// *                  e.g. [{type: 'bar', group_by:'year'}, ...]
+// *   charts_elt_id: id of HTML element to put charts in
+// *   filters_elt_id: id of HTML element to put filter checkboxes in
+// *   chart_wrapper: HTML for wrapper around chart canvas object
+// *   count_template: HTML for record-count element, containing "{count}"
+// *                   somewhere (which will be replaced with the record count).
+// ********************************************************************
+
+
+var ChartView = function(chart_configs, charts_elt_id, filters_elt_id, chart_wrapper, count_template){
+
+  this.state = {
+    data: null,
+    active_filters: [], //put active filters here
+    charts: [],
```

Comment:
//put charts here

Reply:
I don't think we need any of these comments honestly. If it's just a variable named X and the comment is "put X here"....

Reply:
DUH 🤦‍♂️